### ISSUE TITLE
Why is this plugin uses 5 megabytes of space?

### ISSUE THREAD

**Issue by ScienceDiscoverer** [StanDog/npp-zoomdisabler#4]
Can you, please, try to reduce the sizes of DLL and only include x64 binary? Thanks.

**Comment by ScienceDiscoverer** [comment]
I realise that this is very old plug-in, and there is big doubts that original author will ever fix the DLL sizes issue.

For people that want to srink their plug-ins folder but still want this zoom disabling functionality, you can easily make your own plugin to do this in just a few minutes (provided that you already have plug-in template). Just handle `SCI_ZOOM` message in `beNotified(SCNotification *scn)` function of your plug-in:

```
case SCN_ZOOM:
{
        updateScintillaHwnd();
        //int zoom = (i32)m2scintilla(SCI_GETZOOM);
        SendMessage(scintilla_hwnd, SCI_SETZOOM, 0, 0);
}	return;
```
Like this. The whole gist of this plug-in is that it resets zoom to 0 every time you try to change zoom. You can also get current zoom level, if you want, but it's not really needed.

I will include this functionality in my anaglamation plugin for C++ Notepad IDE I'm making, so stay tuned.
Thanks @StanDog for making this plug-in in the first place! It was very helpful until I noticed it's size =)

**Comment by StanDog** [comment]
Hi,

I am not currently working on my plugins, nor do I plan to update them in the near future. All my currently available repositories are open-source, so feel free to use them or copy my code as you need. The license is always the MIT license (I may have forgotten to add a license to this plugin - I will try to do that today and also lock this repository, which I though I already did). The sample code above is correct. There is not much more to do to "freeze" the zoom. There is also a 64-bit version of this plugin available (which is by the way 3.67 MiB in size uncompressed, not 5). You can just repackage it if you want a smaller archive.

The reason the DLLs might be considered to be larger than usual (I personally don't think 3 or 4 MiB is really that big these days) is because the code is written in Delphi Pascal and compiled with Embarcadero RAD Studio / Delphi, and I always compile my applications and libraries together with all the dependencies they need to run. Many other developers often require the user to install additional shared libraries such as Microsoft .NET or JAVA runtime libraries if they do not already have them on their system. This often results in cryptic error messages being displayed to the user at startup without any instructions on what to do. It also reduces platform compatibility if the libraries are loaded at runtime. That's why this plugin is ~3 MiB in size, even though it contains only a few lines written specifically to disable the zoom functionality in Notepad++. There is no need to worry about any "hidden functionality" inside. But if you feel safer compiling the plugin yourself: There is a free version of Delphi (called "Community Edition") that can be used to compile this code.

Good luck with your plugin 👍
